openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -290,6 +291,13 @@ def _update_extended_profile_if_needed(data, user_profile):
         user_profile.save()
 
 
+def _update_state_if_needed(data, user_profile):
+    # If the country was changed to something other than US, remove the state.
+    if "country" in data and data['country'] != 'US':
```

Comment:
Nit: consider using a Constant here for 'US'.  It should make the code more readable.